python-project-template/.pre-commit-config.yaml.jinja

```diff
@@ -94,14 +94,25 @@ repos:
         # https://pre-commit.com/#top_level-default_language_version
         language_version: python3.10
 {%- endif %}
-{%- if 'ruff' in enforce_style %}
+{%- if 'ruff_lint' in enforce_style %}
   - repo: https://github.com/astral-sh/ruff-pre-commit
     # Ruff version.
     rev: v0.1.3
     hooks:
       - id: ruff
-        name: Format and lint code using ruff
+        name: Lint code using ruff
         types_or: [ python, pyi, jupyter ]
+        args: ["check"]
+{%- endif %}
+{%- if 'ruff_format' in enforce_style %}
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    # Ruff version.
+    rev: v0.1.3
+    hooks:
+      - id: ruff
+        name: Format code using ruff
+        types_or: [ python, pyi, jupyter ]
+        args: ["format", "--check"]
 {%- endif %}
```

Comment:
I'm not sure why the hook's readme proposes to put hooks in this order. If we do this, the linter would fail for some bad formatting (for example long lines), before formatter runs. I suggest to swap the order, so formatter runs first and fixes some linting issues

Reply:
I think I agree with the ruff folks on this one!

Since the pre-commit is mostly about warning, and not about fixing the changes, I'm a fan of the lint-first-then-format ordering. This would also alert sooner about "code smell" type problems, before alerting about more minor formatting issues.

Reply:
I'll move this discussion to a new issue. Thank you!